### DISCUSSION TITLE
Use default port for multicast locators [13549]

### DISCUSSION THREAD

**Discussion by omertal88** [eProsima/Fast-DDS#2426] [Q&A]
Hey there,

Is there a way to use the default multicast port without actually specifying it in the profile XML file?
For example for setting up a `metatrafficMulticastLocatorList`, or `initialPeers`.

This port depends on the domain ID, so I figured there should be a way avoid specifying the port and let the parser calculate it for us.

Is that an option? Couldn't find anything on the documentation.

**Comment by JLBuenoLopez** [comment]
Hi @omertal88,

I have moved your question to the proper section in our repository, as this is not an issue with the library but a support question about how to configure and use Fast DDS.

Concerning your question you can just set the IP address of your metatraffic multicast locator. Then, the assigned port to the locator will be 0. And our [documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/listening_locators.html#well-known-ports) states that "Well-known ports are also selected automatically by *Fast DDS* when a Locator is configured with port number *0*".

I hope this helps answering your question!

**Reply by omertal88** [reply]
Yes actually that was very helpful. I'm not sure why it didn't work when I tried.
I think I may have just created an empty `<locator/>` and expected it to work.
I now realize I couldn't leave the multicast IP address unfilled.

I did however, found that UnicastLocators can receive `0.0.0.0` to be used as "my ip". Is that correct?

**Reply by JLBuenoLopez** [reply]
Yes, it is correct. For unicast locators, the IP address `0.0.0.0` is understood as `any`. Then, Fast DDS inquires the kernel asking for the available network interfaces and assigns one locator per interface in the well-known port. However, this is the builtin behavior so I do not see why a user will set this case in the XML file.
